### PR TITLE
Add unit testcases for ensureExternalLoadBalancer to make sure it doesn't panic when errors raised.

### PR DESCRIPTION
**Pull request by grayluck**
**What this PR does / why we need it**:
Add unit testcases for ensureExternalLoadBalancer to make sure it doesn't panic when errors raised. Increase code coverage from 76.5% to 81.5%.

**Release note**: 

```release-note
NONE
```

\assign @MrHohn

### CODE REVIEW

File: pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go [hunk 1]
```diff
@@ -953,3 +953,98 @@ func TestEnsureExternalLoadBalancerDeletedSucceedsOnXPN(t *testing.T) {
...
+		params.clusterID,
+		params.apiService,
+		params.existingFwdRule,
+		params.nodes,
+	)
+	assert.Error(t, err)
+	assert.Nil(t, status)
+}
+
+func TestEnsureExternalLoadBalancerErrors(t *testing.T) {
```
Comment (MrHohn) [inline pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go]: Maybe add a comment explaining this is testing that "it doesn't panic when errors raised"? That is not obvious.

File: pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go [hunk 2]
```diff
@@ -953,3 +953,98 @@ func TestEnsureExternalLoadBalancerDeletedSucceedsOnXPN(t *testing.T) {
...
+
+	params = newEnsureELBParams(nodes)
+	c.MockTargetPools.InsertHook = mock.InsertTargetPoolsInternalErrHook
+	callEnsureExternalLoadBalancerAndCheckError(t, gce, params)
+	c.MockTargetPools.InsertHook = nil
+
+	params = newEnsureELBParams(nodes)
+	c.MockForwardingRules.InsertHook = mock.InsertForwardingRulesInternalErrHook
+	callEnsureExternalLoadBalancerAndCheckError(t, gce, params)
+	c.MockForwardingRules.InsertHook = nil
```
Comment (MrHohn) [inline pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go]: Would be great to group above testcases into a for loop :)

**Review by MrHohn — COMMENTED** [review]
Looks straightforward, don't have many comments.

**Comment by MrHohn** [comment]
cc @@agau4779

File: pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go [hunk 3]
```diff
@@ -953,3 +953,98 @@ func TestEnsureExternalLoadBalancerDeletedSucceedsOnXPN(t *testing.T) {
 	msg := fmt.Sprintf("%s %s %s", v1.EventTypeNormal, eventReasonManualChange, eventMsgFirewallChange)
 	checkEvent(t, recorder, msg, true)
 }
+
+type EnsureELBParams struct {
+	clusterName, clusterID string
```
[the remark this answers is not in the document]
Comment (grayluck) [reply pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go]: Done.

File: pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go (hunk 1 above)
Comment (grayluck) [reply pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go]: Comment added. Vague function name indeed. Let me know if you have a better name for the function in you mind.

File: pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go (hunk 2 above)
Comment (grayluck) [reply pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go]: I tried but I can't figure out a way to unify the mocks - they are all so different.

Comment (MrHohn) [reply pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go]: We can do something flexible in this case (feel free to ignore if it doesn't make sense...):
```golang
	testCases := []struct {
		desc string
		adjustParams func(*EnsureELBParams)
		injectMock func(*cloud.MockGCE)
		resetMock func(*cloud.MockGCE)
	}{...}

	for _, tc := range() {
		params = newEnsureELBParams(nodes)
		if tc.adjustParams != nil {
			tc.adjustParams(params)
		}
		if tc.injectMock != nil {
			tc.injectMock(c)
		}
		callEnsureExternalLoadBalancerAndCheckError(t, gce, params)
		if tc.resetMock != nil {
			tc.resetMock(c)
		}
	}
``

**Review by MrHohn — APPROVED** [review]
Thanks!

/lgtm

**Comment by grayluck** [comment]
Thanks for review Zihong!
/assign @bowei

**Pull request merged**